### PR TITLE
VOQ: Set the ECMP group size to 128.

### PR DESCRIPTION
**Pull request by deepak-singhal0408**
**What I did**
Set the ECMP group size on VOQ platform to 128.

**Why I did it**
in SAI 11.x, the MAX ECMP group size on DNX chipsets is set to 64. This is causing issues on VOQ platforms, where ECMP group size in control plane (ebgp+ibgp) is set to 128. We see ECMP group creation failures and eventually leading to orchagent crash.
Related issue: https://github.com/sonic-net/sonic-buildimage/issues/20622

**How I verified it**
Verified on T2 Chassis(VOQ) by adding Static Routes with > 64 nexthops.
Also tried shutting down on of the group member to verify.

**Details if related**
Syslogs:
NOTICE swss0#orchagent: :- RouteOrch: Switch Type: voq, Max ECMP Group Size supported: 512
 NOTICE swss0#orchagent: :- RouteOrch: Set switch attribute ECMP member count to 128

### CODE REVIEW

File: orchagent/routeorch.cpp [hunk 1]
```diff
@@ -88,6 +89,36 @@ RouteOrch::RouteOrch(DBConnector *db, vector<table_name_with_pri_t> &tableNames,
 
     SWSS_LOG_NOTICE("Maximum number of ECMP groups supported is %d", m_maxNextHopGroupCount);
 
+    /* fetch the MAX_ECMP_MEMBER_COUNT and for voq platform, set it to 128 */
+    attr.id = SAI_SWITCH_ATTR_MAX_ECMP_MEMBER_COUNT;
+    status = sai_switch_api->get_switch_attribute(gSwitchId, 1, &attr);
+    SWSS_LOG_NOTICE("Switch Type: %s, Max ECMP Group Size supported: %d", gMySwitchType.c_str(), attr.value.u32);
```
Comment (prsunny) [inline orchagent/routeorch.cpp:95]: This log should be after the check for `status `as SUCCESS

File: orchagent/routeorch.cpp [hunk 2]
```diff
@@ -88,6 +89,36 @@ RouteOrch::RouteOrch(DBConnector *db, vector<table_name_with_pri_t> &tableNames,
...
+    
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_WARN("Failed to get switch attribute max ECMP Group size. rv:%d", status);
+    }
+    else
+    {
+        uint32_t maxEcmpGroupSize = attr.value.u32;
+        /*If the switch type is voq, and max Ecmp group size supported is greater or equal to 128, set it to 128 */
+        if (gMySwitchType == "voq" && maxEcmpGroupSize >= 128)
```
Comment (prsunny) [inline orchagent/routeorch.cpp:105]: What is the purpose of forcing to 128. This should be ideally based on SAI capability.

Comment (arlakshm) [reply orchagent/routeorch.cpp:105]: Hi @prsunny, in the control plane the max ECMP is set to 128(64 via ebgp and 64 via ibgp learnt routes), the purpose of forcing to 128 here to make sure the control plane and dataplane are in sync

```
 address-family ipv4 unicast
  network 10.1.0.1/32
  network 192.0.0.2/32 route-map HIDE_INTERNAL
  redistribute connected route-map V4_CONNECTED_ROUTES
  neighbor AH_V4 activate
  neighbor AH_V4 soft-reconfiguration inbound
  neighbor AH_V4 maximum-prefix 64000 90 warning-only
  neighbor AH_V4 route-map FROM_AH_V4 in
  neighbor AH_V4 route-map TO_AH_V4 out
  neighbor RH_V4 activate
  neighbor RH_V4 soft-reconfiguration inbound
  neighbor RH_V4 maximum-prefix 64000 90 warning-only
  neighbor RH_V4 route-map FROM_RH_V4 in
  neighbor RH_V4 route-map TO_RH_V4 out
  neighbor VOQ_CHASSIS_V4_PEER activate
  neighbor VOQ_CHASSIS_V4_PEER addpath-tx-all-paths
  neighbor VOQ_CHASSIS_V4_PEER soft-reconfiguration inbound
  neighbor VOQ_CHASSIS_V4_PEER route-map FROM_VOQ_CHASSIS_V4_PEER in
  neighbor VOQ_CHASSIS_V4_PEER route-map TO_VOQ_CHASSIS_V4_PEER out
  maximum-paths 64
  maximum-paths ibgp 64
 exit-address-family
 !
```

File: orchagent/routeorch.cpp (hunk 1 above)
Comment (deepak-singhal0408) [reply orchagent/routeorch.cpp]: Thanks Prince. addressed this.

**Comment by deepak-singhal0408** [comment]
@prsunny, 
Following are the reasons for the missing code coverage.
 1. At present we dont have vslib support for SAI_SWITCH_ATTR_MAX_ECMP_MEMBER_COUNT attribute. Hence the dvs test run doesnt execute the else code. 
 2. This is a platform specific code.
 
In future, once we have vslib support for this attribute, this code will automatically be covered during dvs bring up of virtual chassis (as this will get executed during swss init)

This is to fix a regression on 202405 branch for voq chassis, hence request to expedite its merge. Thank you!

**Comment by bingwang-ms** [comment]
@deepak-singhal0408 Please add MSFT ADO number to the PR for 202405 cherry-pick. Thanks!

**Comment by deepak-singhal0408** [comment]
MSFT ADO: 30182195 for 202405 branch.. @bingwang-ms

**Pull request merged**